include/sys/cred.h

```diff
@@ -53,6 +54,13 @@ int cred_cansignal(proc_t *p, cred_t *cred);
  * with lock held. */
 int proc_cansignal(proc_t *target, signo_t sig);
 
+/* Checks if given gid is in supplementary groups of given credentials set */
+bool cred_groupmember(gid_t gid, cred_t *cred);
+
+/* VFS checks */
+bool cred_can_chmod(uid_t f_owner, gid_t f_group, cred_t *cred, mode_t mode);
```

Comment:
Maybe it'll save you some code if you returned `errno` (like `EPERM`) instead of boolean? Please reevaluate if using standard unix error reporting trick could be better in this case.

Reply:
I don't think so. I use it only in `tmpfs.c:297` in if condition. There is no need to return errorcode because it is only error code possible here.